frontend/src/components/NewRunParametersV2.tsx

```diff
@@ -91,13 +107,45 @@ function convertInput(paramStr: string, paramType: ParameterType_ParameterTypeEn
   }
 }
 
+function checkInput(
+  parametersInRealType: any,
+  paramType: ParameterType_ParameterTypeEnum,
+): { validInput: boolean; errorMessage: string } {
```

Comment:
Usually when we are trying to return an object with multiple fields, we will define an interface which defines the structure of such object first. So this return value can be:

```
interface InputInformation {
   isValidInput: boolean;
   errorMessage: string;
}
```

However, I am wondering if it is needed to have two separate fields. For example: Can we say input is valid when `errorMessage` is empty, and input is invalid when `errorMessage` is not empty? Then you just need to return errorMessage in this case. And you don't need to maintain the following state:

```
const [validInputs, setValidInputs] = useState([]);
```

Reply:
Great suggestion! I have thought this option before, but haven't tried yet. The reason I choose to use two separate fields is making it more clear. However, I think use only 1 field `errorMessage` is more concise!